plugins/Toolbox/resources/qml/ToolboxLicenseDialog.qml

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2018 Ultimaker B.V.
+// Toolbox is released under the terms of the LGPLv3 or higher.
+
+import QtQuick 2.2
+import QtQuick.Dialogs 1.1
+import QtQuick.Window 2.2
+import QtQuick.Controls 1.4
+import QtQuick.Controls.Styles 1.4
+
+// TODO: Switch to QtQuick.Controls 2.x and remove QtQuick.Controls.Styles
+
+import UM 1.1 as UM
+
+UM.Dialog
+{
+    title: catalog.i18nc("@title:window", "Plugin License Agreement")
+    minimumWidth: UM.Theme.getSize("license_window_minimum").width
+    minimumHeight: UM.Theme.getSize("license_window_minimum").height
+    width: minimumWidth
+    height: minimumHeight
+    property var pluginName;
+    property var licenseContent;
+    property var pluginFileLocation;
+    Item
+    {
+        anchors.fill: parent
+        Label
+        {
+            id: licenseTitle
+            anchors.top: parent.top
+            anchors.left: parent.left
+            anchors.right: parent.right
+            text: licenseDialog.pluginName + catalog.i18nc("@label", "This plugin contains a license.\nYou need to accept this license to install this plugin.\nDo you agree with the terms below?")
+            wrapMode: Text.Wrap
+        }
+        TextArea
+        {
+            id: licenseText
+            anchors.top: licenseTitle.bottom
+            anchors.bottom: parent.bottom
+            anchors.left: parent.left
+            anchors.right: parent.right
+            anchors.topMargin: UM.Theme.getSize("default_margin").height
+            readOnly: true
+            text: licenseDialog.licenseContent != null ? licenseDialog.licenseContent : ""
```

Comment:
`licenseContent` can never be `null`, right?

Reply:
I think it can. It typically is None unless there is a license which is only the case with some plugins, but never with materials.

Reply:
Oh you are correct. This check needs to exist though, which is why it's *also* in Python. Removing it here. Fixed in 62aeb6c

Reply:
But if there is no license, the `ToolboxLicenseDialog` can never be shown, I was assuming.